x/common/types/key_rollapp_packet.go

```diff
@@ -27,30 +25,49 @@ var (
 
 // RollappPacketKey constructs a key for a specific RollappPacket
 func RollappPacketKey(rollappPacket *RollappPacket) []byte {
+	rollappIdBytes := []byte(rollappPacket.RollappId)
 	// Get the relevant key prefix based on the packet status
-	statusPrefix := MustGetStatusBytes(rollappPacket.Status)
+	statusBytes := mustGetStatusBytes(rollappPacket.Status)
 	// Build the key bytes repr. Convert each uint64 to big endian bytes to ensure lexicographic ordering.
-	keySeparatorBytes := []byte(KeySeparator)
-	rollappIdBytes := []byte(rollappPacket.RollappId)
+	keySeparatorBytes := []byte(keySeparator)
 	proofHeightBytes := sdk.Uint64ToBigEndian(rollappPacket.ProofHeight)
 	// Build the packetUID from the destination channel and sequence number.
 	packetSequenceBytes := sdk.Uint64ToBigEndian(rollappPacket.Packet.Sequence)
 	packetDestinationChannelBytes := []byte(rollappPacket.Packet.DestinationChannel)
 	packetUIDBytes := append(packetDestinationChannelBytes, packetSequenceBytes...)
 
 	// Concatenate the byte slices directly.
-	result := append(statusPrefix, keySeparatorBytes...)
-	result = append(result, proofHeightBytes...)
-	result = append(result, keySeparatorBytes...)
+	// status/rollappID/proofHeight/packetUID
+	result := append(statusBytes, keySeparatorBytes...)
 	result = append(result, rollappIdBytes...)
 	result = append(result, keySeparatorBytes...)
+	result = append(result, proofHeightBytes...)
+	result = append(result, keySeparatorBytes...)
 	result = append(result, packetUIDBytes...)
 
 	return result
 }
 
-// MustGetStatusBytes returns the byte representation of the status
-func MustGetStatusBytes(status Status) []byte {
+// RollappPacketStatusAndRollappIDKey constructs a key prefix for a specific RollappPacket
+// by status and rollappID:
+// "status/rollappID/"
+// by only status:
+// "status/"
+func RollappPacketStatusAndRollappIDKey(rollappPacket *RollappPacket) (result []byte) {
+	rollappIdBytes := []byte(rollappPacket.RollappId)
+	statusPrefix := mustGetStatusBytes(rollappPacket.Status)
+	keySeparatorBytes := []byte(keySeparator)
+	result = append(statusPrefix, keySeparatorBytes...)
+	if rollappPacket.RollappId == "" {
```

Comment:
I would separate to `RollappPacketByStatusKey`
and call it `IteratorKey`, as it's key expected to use as iterator